llvm/lib/Analysis/InlineOrder.cpp

```diff
@@ -216,14 +276,43 @@ class PriorityInlineOrder : public InlineOrder<std::pair<CallBase *, int>> {
     };
   }
 
+  PriorityInlineOrder(FunctionAnalysisManager &FAM, const InlineParams &Params,
```

Comment:
I am wondering if it would make more sense to create a new subclass of InlineOrder instead of modifying this existing one.

Reply:
e.g., `class TopDownInlineOrder : public InlineOrder<std::pair<CallBase *, int>>`

Reply:
then you don't need to touch any of the others

Reply:
I feel like it would add a lot of repeted code for little gain, we'd basically have to make a carbon copy of the exisitng PriorityInlineOrder with our new constructor. I'm not sure if would be worth it.